File: 49.group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> groups;
        for (auto &str : strs) {
            string key = str;
            sort(key.begin(), key.end());
            groups[key].push_back(str);
        }
        vector<vector<string>> ans;
        for (auto& p : groups) {
            ans.push_back(p.second);
        }
        return ans;
    }
};

File: 49.group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let groups = {};
    for (let i = 0; i < strs.length; i++) {
        let str = strs[i];
        let key = str.split("").sort().join("");
        if (groups[key]) {
            groups[key].push(str);
        } else {
            groups[key] = [str];
        }
    }
    let ans = [];
    for (let key in groups) {
        ans.push(groups[key]);
    }
    return ans;
};